src/spyglass/common/common_ephys.py

```diff
@@ -812,6 +812,10 @@ def make(self, key):
         # load in the timestamps
         timestamps = np.asarray(lfp_object.timestamps)
 
+        # leave nan timestamps out. Raw nwb files can have small amount (a couple samples) of nan.
+        notnan_mask = ~np.isnan(timestamps)
```

Comment:
Is there any way we can parameterize this in the firfilterparams table to maintain it as an option?

Reply:
The problem is that if there is NaN in the timestamps, data won't even go through some of the functions.

Reply:
If previous data did not bug at this step (as it happens really rarely, and NaN's likely got intersected out ), the code effectively does not exist even when these lines are added.

Reply:
That doesn't answer my question about parameterization, which would support maintaining records of which files did vs. did not need this adjustment. 

If the answer is no, a short term custom table should be able to tell us whether or not existing files would be have been processed otherwise by this change

```python
import datajoint as dj
import numpy as np

from spyglass.common.common_ephys import LFP, LFPBandSelection # noqa: F401

schema = dj.schema(f"{dj.config['database.user']_temp")
@schema
class WhichNans(dj.Computed):
    definition = """
    -> LFPBandSelection
    ---
    has_nans : bool
    """

    def make(self, key):
        nwb_dict = {'nwb_file_name': key['nwb_file_name']}
        lfp_object = (LFP & nwb_dict).fetch_nwb()[0]['lfp'] 
        has_nans = np.any(np.isnan(np.asarray(lfp_object.timestamps))
```

Reply:
The following would be changed by this edit

```python
have_nans = {
    "J1620230614_.nwb",
    "eliot20221017_.nwb",
    "eliot20221020_.nwb",
    "eliot20221024_.nwb",
    "eliot20221026_.nwb",
    "molly20220418_.nwb",
    "molly20220420_.nwb",
}
```

We can either ...
1. Recompute all these keys, and everything downstream (including `jguidera_lfp.__epoch_theta_l_f_p`)
2. Add a secondary attribute to `FirFilterParameters` to determine this behavior
3. Find a place in your custom analysis to fix this issue for your data, outside of spyglass

Do you have a preference from these options, @edeno ?

Reply:
I think based on conversation with @samuelbray32, the third option seems like the easiest here because of the way things are implemented. Is that right @samuelbray32 ?

Reply:
Yes, the implementation for the custom downstream code would allow for data to come from a separate table without dropping existing entries.

Do we know if the `nan` timestamp values are present in the original nwb file or introduced by spyglass?

Reply:
I don't know either way - I only looked to the `LFP` table where this data is fetched from

Reply:
@shijiegu ?

Reply:
- I checked the raw nwb for `eliot20221017_.nwb` and the `nan` timestamp exists in the acquisition e-series. 
- Given that this file and \all the affected ones Chris identified are older, I'd guess the nan value stems from conversion by `rec_to_nwb` 
- In terms of a spyglass package solution, I think we should consider a nan timestamp in the raw data as an error (rather than try assume an interpolation)
    - In that case it may make more sense to use intervals which exclude nan time values in `LFP`
    - Alternatively, could check for nan time values during data insert and raise error then instead

Reply:
@samuelbray32 I like the solution where we use the interval list to exclude the NaNs.

@shijiegu can you reprocess these datasets with an IntervalList time that excludes the NaN at the beginning? That is how the code is designed and would help keep track of the exclusion.